scripts/mkcloud

```diff
@@ -670,38 +678,27 @@ function restartcloud
 }
 
 # bring up VMs that will take cloud controller/compute/storage roles
-function setupnodes
+function get_nodenumbercontroller
 {
-    onadmin wait_tftpd || return $?
-
     local nodenumbercontroller=1
     if [[ $clusterconfig == *services* ]]; then
         nodenumbercontroller=`echo ${clusterconfig}\
             | sed -e "s/^.*services[^:]*=\([[:digit:]]\+\).*/\1/"`
     fi
+    echo $nodenumbercontroller
+}
+
+
+# bring up VMs that will take cloud controller/compute/storage roles
+function setupnodes
+{
+    onadmin wait_tftpd || return $?
+    local nodenumbercontroller=`get_nodenumbercontroller`
 
     setuppublicnet
     for i in $(nodes ids normal) ; do
-        local macaddress=$(macfunc $i)
-
-        # transport drdb volume information to admin node (needed for proposal of data cluster)
-        drbd_serial=""
-        if [ $drbd_hdd_size != 0 ]; then
-            if [ $i -le 2 ] ; then
-                drbd_serial="$cloud-node$i-drbd"
-                # libvirt does not accept anything other than [:alnum:]_-
-                # for serial strings:
-                drbd_serial=${drbd_serial//[^A-Za-z0-9-_]/_}
-                drbdnode_mac_vol="${drbdnode_mac_vol}+${macaddress}#${drbd_serial}"
-                drbdnode_mac_vol="${drbdnode_mac_vol#+}"
-            fi
-        fi
-
-        ${mkcloud_lib_dir}/libvirt/compute-config $cloud $i $macaddress\
-            $controller_raid_volumes $cephvolumenumber "$drbd_serial"\
-            $compute_node_memory $controller_node_memory $libvirt_type $vcpus\
-            $emulator $vdisk_dir 3 $nodenumbercontroller > /tmp/$cloud-node$i.xml
-        ${mkcloud_lib_dir}/libvirt/vm-start /tmp/$cloud-node$i.xml
+        ${mkclouddriver}_do_onhost_create_compute_config $i
+        ${mkcloud_lib_dir}/${mkclouddriver}/vm-start /tmp/$cloud-node$i.xml
```

Comment:
No, setupnodes != setuplonelynodes
Just the creation of the libvirt xml is unified.